demos/python_demos/common/models/hpe_associative_embedding.py

```diff
@@ -17,6 +17,94 @@
+        if aspect_ratio >= 1.0:  # img width >= height
+            input_height, input_width = target_size, round(target_size * aspect_ratio)
+        else:
+            input_height, input_width = round(target_size / aspect_ratio), target_size
+        self.h = (input_height + size_divisor - 1) // size_divisor * size_divisor
+        self.w = (input_width + size_divisor - 1) // size_divisor * size_divisor
+        default_input_shape = self.net.input_info[self.image_blob_name].input_data.shape
+        input_shape = {self.image_blob_name: (default_input_shape[:-2] + [self.h, self.w])}
+        self.logger.info('Reshape net to {}'.format(input_shape))
+        self.net.reshape(input_shape)
+
+        self.decoder = AssociativeEmbeddingDecoder(
+            num_joints=self.net.outputs[self.heatmaps_blob_name].shape[1],
+            adjust=True,
+            refine=True,
+            delta=0.0,
+            max_num_people=30,
+            detection_threshold=0.1,
+            tag_threshold=1,
+            pose_threshold=prob_threshold,
+            use_detection_val=True,
+            ignore_too_much=False)
+
+    @staticmethod
+    def _get_inputs(net):
+        image_blob_name = None
+        for blob_name, blob in net.input_info.items():
+            if len(blob.input_data.shape) == 4:
+                image_blob_name = blob_name
+            else:
+                raise RuntimeError('Unsupported {}D input layer "{}". Only 2D and 4D input layers are supported'
+                                   .format(len(blob.shape), blob_name))
+        if image_blob_name is None:
+            raise RuntimeError('Failed to identify the input for the image.')
+        return image_blob_name
+
+    def preprocess(self, inputs):
+        img = resize_image(inputs, (self.w, self.h), keep_aspect_ratio=True)
+        h, w = img.shape[:2]
+        meta = {'original_shape': inputs.shape,
+                'resized_shape': img.shape}
+        img = np.pad(img, ((0, self.h - h), (0, self.w - w), (0, 0)),
+                     mode='constant', constant_values=0)
+        img = img.transpose((2, 0, 1))  # Change data layout from HWC to CHW
+        img = img[None]
+        return {self.image_blob_name: img}, meta
+
+    def postprocess(self, outputs, meta):
+        heatmaps = outputs[self.heatmaps_blob_name]
+        nms_heatmaps = outputs[self.nms_heatmaps_blob_name]
+        aembds = outputs[self.embeddings_blob_name]
+        poses, scores = self.decoder(heatmaps, aembds, nms_heatmaps=nms_heatmaps)
+        # Rescale poses to the original image.
+        original_image_shape = meta['original_shape']
+        resized_image_shape = meta['resized_shape']
+        scale_x = original_image_shape[1] / resized_image_shape[1]
+        scale_y = original_image_shape[0] / resized_image_shape[0]
+        poses[:, :, 0] *= scale_x * self.output_scale
+        poses[:, :, 1] *= scale_y * self.output_scale
+        return poses, scores
+
+
+def find_layer_by_name(name, layers):
```

Comment:
I think this function should be removed to `Model` class as method, since it is used elsewhere.

Reply:
This one has a different implementation.